src/main/java/com/hopper/cloud/airlines/HopperPaymentClient.java

```diff
@@ -63,7 +74,52 @@ public String tokenizePaymentCard(PaymentCardDetail paymentCardDetail) throws Ap
         if (response.getStatus() == 201) {
             return response.getBody().getTransaction().getPaymentMethod().getToken();
         } else {
-            throw new ApiException("Unable to create this specific token, response : " + response.getStatus());
+            throw new ApiException("Unable to create a token, response : " + response.getStatus());
+        }
+    }
+
+    /**
+     * Perform credit card tokenization by encrypting some fields
+     * @param creditCardDetail
+     * @return The token if it was successfully retrieved, else null
+     */
+    public String tokenizePaymentCreditCardWithEncryption(CreditCardDetail creditCardDetail) {
+        String token = null;
+        if (StringUtil.isEmpty(encryptionKeyId) || StringUtil.isEmpty(encryptionPublicKey)) {
+            // Log the error message
+            //logger.error("Missing Encryption parameters for payment");
```

Comment:
If the encryption parameters are missing, can we proceed with tokenization but without encrypting?

Reply:
Good catch !
Previously we threw an exception in case encryption params were missing, but since it is no longer the case, we must ensure that we won't perform tokenization if they are missing